packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -324,118 +313,88 @@ class DevFS {
     return _operations.destroy(fsName);
   }
 
-  void _reset() {
-    // Reset the dirty byte count.
-    _bytes = 0;
-    // Mark all entries as possibly deleted.
-    _entries.forEach((String path, DevFSEntry entry) {
-      entry._exists = false;
-    });
-    // Clear the dirt entries list.
-    _dirtyEntries.clear();
-    // Clear the deleted entries list.
-    _deletedEntries.clear();
-    // Clear the dirty asset entries.
-    dirtyAssetEntries.clear();
-  }
-
-  Future<dynamic> update({ DevFSProgressReporter progressReporter,
+  /// Update files on the device and return the number of bytes sync'd
+  Future<int> update({ DevFSProgressReporter progressReporter,
                            AssetBundle bundle,
                            bool bundleDirty: false,
                            Set<String> fileFilter}) async {
-    _reset();
+    // Mark all entries as possibly deleted.
+    _entries.forEach((String devicePath, DevFSContent content) {
```

Comment:
you could also write:

for (DevFSContent content in _entries.values) {
  content._exists = false;
}